### PR TITLE
PXP-9416: Prevent slack webhook from exiting out

### PR DESCRIPTION
**Pull request by MaribelleHGomez**
Replace system exit with log in the event of an error
Use slack sdk for webhook calls, could potentially build out more elaborate slack notifications in the future

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### New Features
- Implemented error log, over system exit on webhook fail

### Dependency updates
- slack-sdk added

### CODE REVIEW

**Review by frankliuao — APPROVED** [review]
Using slack sdk and utilizing AssertionError to debug. Straightforward to implement. Approved.

File: replicate.py [hunk 1]
```diff
@@ -102,6 +108,8 @@ def parse_arguments():
...
 
     try:
-        req = requests.post(SLACK_URL, json={"text": f"Completed {args.action}"})
-    except requests.exceptions.RequestException as e:
-        raise SystemExit(e)
+        slack_call = webhook.send(text=f"Completed {args.action}")
+        assert slack_call.status_code == 200
+        assert slack_call.body == "ok"
+    except AssertionError as e:
+        logger.error("The slack hook has an encountered an error: Detail {}".format(e))
```
Comment (BinamB) [inline replicate.py:115]:
```suggestion
        logger.error("The slack hook has encountered an error: Detail {}".format(e))
```

File: replicate.py [hunk 2]
```diff
@@ -60,11 +63,14 @@ def parse_arguments():
...
     args = parse_arguments()
+    webhook = WebhookClient(SLACK_URL)
 
     try:
-        req = requests.post(SLACK_URL, json={"text": f"Starting {args.action}"})
-    except requests.exceptions.RequestException as e:
-        raise SystemExit(e)
+        slack_call = webhook.send(text=f"Starting {args.action}")
+        assert slack_call.status_code == 200
+        assert slack_call.body == "ok"
```
Comment (BinamB) [inline replicate.py:71]: I think just the `status_code` assert should be fine.

File: replicate.py [hunk 3]
```diff
@@ -102,6 +108,8 @@ def parse_arguments():
     print("Total time: {} seconds".format(end - start))
 
     try:
-        req = requests.post(SLACK_URL, json={"text": f"Completed {args.action}"})
-    except requests.exceptions.RequestException as e:
-        raise SystemExit(e)
+        slack_call = webhook.send(text=f"Completed {args.action}")
+        assert slack_call.status_code == 200
+        assert slack_call.body == "ok"
```
Comment (BinamB) [inline replicate.py:113]: same here. `status_code` assertion should be fine

File: replicate.py (hunk 1 above)
Comment (frankliuao) [reply replicate.py:115]: Nice catch on the grammar, @BinamB !

**Review by BinamB — APPROVED** [review]

**Pull request merged**